store/datastore/utils.go

```diff
@@ -38,34 +39,33 @@ func rebind(query string) string {
 
 // helper function that converts a simple repsitory list
 // to a sql IN statment.
-func toList(listof []*model.RepoLite) (string, []interface{}) {
-	var size = len(listof)
-	if size > 999 {
-		size = 999
-		listof = listof[:999]
-	}
-	var qs = make([]string, size, size)
-	var in = make([]interface{}, size, size)
-	for i, repo := range listof {
-		qs[i] = "?"
-		in[i] = repo.FullName
-	}
-	return strings.Join(qs, ","), in
-}
+func toList(listof []*model.RepoLite) ([]string, [][]interface{}) {
+	var total = len(listof)
```

Comment:
> It did not seem attractive for me because the limit variable

this limit could be a global constant, for example:

```
const maxPlaceholder = 999
```

> core logic for pagination was replicated

I think this is OK

In some cases duplicate logic is not bad. For example, the reason that I split `toList` and `toListPostgres` into separate functions was that it was more efficient. The `toListPostgres` function has to modify the length of the query string resulting in extra allocations for string manipulation. So by combining into a single function (while well intentioned) it removed a performance optimization.